spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/reactive/PubSubReactiveFactory.java

```diff
@@ -72,23 +72,29 @@ public PubSubReactiveFactory(PubSubSubscriberOperations subscriberOperations, Sc
 	 * batches of up to the requested number of messages until the full demand is fulfilled
 	 * or subscription terminated.
 	 * <p>For unlimited demand, the underlying subscription will be polled at a regular interval,
-	 * requesting up to {@code Integer.MAX_VALUE} messages at each poll.
+	 * requesting up to {@code maxMessages} messages at each poll.
 	 * <p>For specific demand, as many messages as are available will be returned immediately,
 	 * with remaining demand being fulfilled in the future.
 	 * Pub/Sub timeout will cause a retry with the same demand.
 	 * <p>Any exceptions that are thrown by the Pub/Sub client will be passed as an error to the stream.
 	 * The error handling operators, like {@link Flux#retry()},
 	 * can be used to recover and continue streaming messages.
 	 * @param subscriptionName subscription from which to retrieve messages.
-	 * @param pollingPeriodMs how frequently to poll the source subscription in case of unlimited demand, in milliseconds.
+	 * @param pollingPeriodMs how frequently to poll the source subscription in case of
+	 * unlimited demand, in milliseconds.
+	 * @param maxMessages max number of messages that may be pulled from the source
+	 * subscription in case of unlimited demand.
 	 * @return infinite stream of {@link AcknowledgeablePubsubMessage} objects.
+	 * @throws IllegalArgumentException if {@code maxMessages < 0}
 	 */
-	public Flux<AcknowledgeablePubsubMessage> poll(String subscriptionName, long pollingPeriodMs) {
-
+	public Flux<AcknowledgeablePubsubMessage> poll(String subscriptionName, long pollingPeriodMs, int maxMessages) {
```

Comment:
Methods with multiple numeric parameters can get confusing (we have `String`, `long`, `int` here). I think I'd prefer to set `maxMessages` in an overloaded `PubSubReactiveFactory` constructor. 

Upsides:
* we get to keep a single `poll()` method. 
* `maxMessages` can be autoconfigured to current max int value by default and overridden with a property, keeping it backwards compatible and consistent with how we configure other functionality.
* `maxMessages` won't need to be propagated from `poll()` to `pollingPull()` to `pullAll()`.

Downside:
* different instances of the factory have to be created if different `maxMessages` settings are required in a single application. This is mitigated by how lightweight `PubSubReactiveFactory` is; it's just a wrapper around `PubSubTemplate` and a reactor scheduler.

@meltsufin What do you think?

Reply:
Makes sense, I see the `maxMessages` setting being something not changed too frequently so setting in constructor seems appropriate.

Reply:
Overloaded constructor or overloaded `poll` methods would work. Since it affects multiple methods, constructor is probably preferred.
Either way, the way it is now, is not good because it would causes a breaking change.